### PR TITLE
320220831_ajout_sensitivity

### PR DESCRIPTION
**Pull request by MaelleTtrt**
On donne la possibilité à l'utilisateur de changer la sensibilité.

### CODE REVIEW

File: source/qsub_spectroGenerationFileScale.py [hunk 1]
```diff
@@ -32,9 +32,13 @@ def generate_and_save_figures(colmapspectros, segment_times, Freq, log_spectro,
     fact_y = 1.3
     fig, ax = plt.subplots(nrows=1, ncols=1, figsize=(fact_x * 1800 / my_dpi, fact_y * 512 / my_dpi), dpi=my_dpi)
     color_map = plt.cm.get_cmap(colmapspectros)  # .reversed()
+    print('LOG ', np.mean(log_spectro))
```
Comment (cazaudo) [inline source/qsub_spectroGenerationFileScale.py:35]: to be removed , same for the following l 36 37 ...

File: source/qsub_spectroGenerationFileScale.py [hunk 2]
```diff
@@ -93,14 +111,17 @@ def gen_spectro(max_w, min_color_val, data, sample_rate, win_size, nfft, pct_ove
...
 
 
-def gen_tiles(nber_level, data, sample_rate, output, winsize, nfft, overlap, min_color_val, segment_str):
-    if not isnorma:
-        data = (data - np.mean(data)) / np.std(data)
-    else:
-        data = (data - zscore_mean) / zscore_std
 
-    duration = len(data) / int(sample_rate)
+def gen_tiles(nber_level, data, sample_rate, output, winsize, nfft, overlap, min_color_val, segment_str, scaling='density'):
```
Comment (cazaudo) [inline source/qsub_spectroGenerationFileScale.py:115]: how do you justify the scaling names of 'density' and 'spectrum' ?

File: source/qsub_spectroGenerationFileScale.py [hunk 3]
```diff
@@ -147,40 +171,47 @@ def gen_tiles(nber_level, data, sample_rate, output, winsize, nfft, overlap, min
...
     segment_str = ''
 
     if nberAdjustSpectros == 0:
-        data, sample_rate = soundfile.read(os.path.join(path_audio_files, audio_file))
-
+        data, sample_rate = soundfile.read(os.path.join(path_audio_files, audio_file))        
+        # add sensitivity
+        data =(data*peak_voltage)/sensitivity
+        
+        print('DATA MEAN : ', np.mean(data**2))
```
Comment (cazaudo) [inline source/qsub_spectroGenerationFileScale.py:190]: to be removed

File: source/qsub_spectroGenerationFileScale.py [hunk 4]
```diff
@@ -212,34 +243,43 @@ def process_file(audio_file):
...
-
-        nber_wav_to_be_processed = 4
+        
+        # Parameters used to  the real dB value of the spectrogram
+        peak_voltage = 2.5    # Peak voltage in volt
+        # nb_bits = 16   # number of bits of the coded signal
+        # datatype = 'int16'
+        sensitivity_dB = -163.5  # Sensitivity of the hydrophone in dB ref 1µPa
+        sensitivity = 10**(sensitivity_dB/20) * 1e6
+        nber_wav_to_be_processed = 1
```
Comment (cazaudo) [inline source/qsub_spectroGenerationFileScale.py:274]: please add lines 267 to 274 after the parameter fmin_HighPassFilter below , so your paramaters can be used in  the Datarmor version

File: source/qsub_spectroGenerationFileScale.py [hunk 5]
```diff
@@ -377,22 +414,24 @@ def process_file(audio_file):
...
     list_wav_withEvent = [os.path.basename(x) for x in list_wav_withEvent]
 
+
     if nberAdjustSpectros==0:
  
-        ncpus = 10
-        with Pool(processes=ncpus) as pool:
-            pool.map(process_file, list_wav_withEvent)
-            pool.close()
+        # ncpus = 10
```
Comment (cazaudo) [inline source/qsub_spectroGenerationFileScale.py:423]: please uncomment the multithread version (should be used by default) and add input parameters peak_voltage, sensitivity to process_file

**Review by cazaudo — CHANGES_REQUESTED** [review]

File: source/qsub_spectroGenerationFileScale.py (hunk 2 above)
Comment (MaelleTtrt) [reply source/qsub_spectroGenerationFileScale.py:115]: Ce sont les termes utilisés dans SciPy

File: source/qsub_spectroGenerationFileScale.py (hunk 4 above)
Comment (MaelleTtrt) [reply source/qsub_spectroGenerationFileScale.py:274]: I added the parameters sensitivity_dB, peak_voltage and scaling after the parameter fmin_HighPassFilter

File: source/qsub_spectroGenerationFileScale.py (hunk 5 above)
Comment (MaelleTtrt) [reply source/qsub_spectroGenerationFileScale.py:423]: parameters peak_voltage, sensitivity are already added to process_file

**Pull request merged**